src/apps/proj_strtod.cpp

```diff
@@ -123,6 +125,13 @@ double proj_strtod(const char *str, char **endptr) {
         return 0;
     }
 
+    /* NaN */
+    if (strncmp(p, "NaN", 3) == 0) {
```

Comment:
should this comparison also include `nan`?

Reply:
Technically, if we're going for feature parity with standard strtod, then it should handle `nan`, `NAN`, `NaN`, and maybe every combination of upper and lowercase. I mentioned in a higher up comment:

> I'm not sure how strict/flexible this code needs to be. It specifically checks for only NaN (not "nan", not "NAN", etc). It also doesn't do anything if the character after NaN is a whitespace. For a valid NaN it should be, but right now someone could technically enter the next coordinate number right after a NaN (ex. NaN0.0m would be two values I think).

Reply:
you may use the ci_equal() function for case insensitive comparison, provided you add at top of file
```
#include "proj/internal/internal.hpp"
using namespace NS_PROJ::internal;
```

Note that GDAL's strtod() function also catches other variations of NaN that used to be/are output by Microsoft C runtime library. Cf https://github.com/OSGeo/gdal/blob/master/port/cpl_strtod.cpp#L227 and below

Reply:
ah, actually ci_equal() isn't appropriate. That should be ci_starts_with()

Reply:
How may those comparisons affect performance?

Reply:
```cpp
#include "proj/internal/internal.hpp"
using namespace NS_PROJ::internal;
```

results in:

```
In file included from /home/davidh/repos/git/PROJ/src/apps/proj_strtod.cpp:89:
/home/davidh/repos/git/PROJ/include/proj/internal/internal.hpp:30:2: error: #error This file should only be included from a PROJ cpp file
   30 | #error This file should only be included from a PROJ cpp file
      |  ^~~~~
gmake[2]: *** [src/apps/CMakeFiles/cct.dir/build.make:90: src/apps/CMakeFiles/cct.dir/proj_strtod.cpp.o] Error 1
```

Reply:
add #define FROM_PROJ_CPP before including proj/internal/internal.hpp

Reply:
```c
#include "proj_strtod.h"
#define FROM_PROJ_CPP
#include "proj/internal/internal.hpp"
using namespace NS_PROJ::internal;

#include <ctype.h>
#include <errno.h>
```

It's been a long time since I've done real C, and I really don't know C++ and especially not namespaces...

```c
    if (ci_starts_with(p, "NaN")) {
```

```
/usr/bin/ld: CMakeFiles/cct.dir/proj_strtod.cpp.o: in function `proj_strtod(char const*, char**) [clone .part.0]':
proj_strtod.cpp:(.text+0x6b): undefined reference to `osgeo::proj::internal::ci_starts_with(char const*, char const*)'
```

Reply:
> proj_strtod.cpp:(.text+0x6b): undefined reference to `osgeo::proj::internal::ci_starts_with(char const*, char const*)'

ah, I missed proj_strtod.cpp was an application file, and not a file in the library. ci_starts_with() needs to be exported then.

You need to apply the following patch:
```
diff --git a/include/proj/internal/internal.hpp b/include/proj/internal/internal.hpp
index dda11cbc..d7aa0d49 100644
--- a/include/proj/internal/internal.hpp
+++ b/include/proj/internal/internal.hpp
@@ -128,7 +128,7 @@ inline bool starts_with(const std::string &str, const char *prefix) noexcept {
 
 bool ci_less(const std::string &a, const std::string &b) noexcept;
 
-bool ci_starts_with(const char *str, const char *prefix) noexcept;
+PROJ_DLL bool ci_starts_with(const char *str, const char *prefix) noexcept;
 
 bool ci_starts_with(const std::string &str, const std::string &prefix) noexcept;
 
diff --git a/scripts/reference_exported_symbols.txt b/scripts/reference_exported_symbols.txt
index 6ff75bef..fc512037 100644
--- a/scripts/reference_exported_symbols.txt
+++ b/scripts/reference_exported_symbols.txt
@@ -335,6 +335,7 @@ osgeo::proj::HorizontalShiftGridSet::reopen(pj_ctx*)
 osgeo::proj::internal::ci_equal(std::string const&, char const*)
 osgeo::proj::internal::ci_equal(std::string const&, std::string const&)
 osgeo::proj::internal::ci_find(std::string const&, char const*)
+osgeo::proj::internal::ci_starts_with(char const*, char const*)
 osgeo::proj::internal::c_locale_stod(std::string const&)
 osgeo::proj::internal::replaceAll(std::string const&, std::string const&, std::string const&)
 osgeo::proj::internal::split(std::string const&, char)
```